src/ripple/app/main/Application.cpp

```diff
@@ -1035,7 +999,33 @@ class ApplicationImp : public Application, public RootStoppable, public BasicApp
                 return validators().trustedPublisher(pubKey);
             });
 
-        stopped();
+        // The order of these stop calls is delicate.
+        // Re-ordering them risks undefined behavior.
+        m_loadManager->stop();
+        m_shaMapStore->stop();
+        m_jobQueue->stop();
+        if (shardArchiveHandler_)
+            shardArchiveHandler_->stop();
+        if (overlay_)
+            overlay_->stop();
+        if (shardStore_)
+            shardStore_->stop();
+        grpcServer_->stop();
+        m_networkOPs->stop();
+        serverHandler_->stop();
+        m_ledgerReplayer->stop();
+        m_inboundTransactions->stop();
+        m_inboundLedgers->stop();
+        ledgerCleaner_->stop();
+        if (config_->reporting())
+        {
+            reportingETL_->stop();
+            dynamic_cast<RelationalDBInterfacePostgres*>(
+                &*mRelationalDBInterface)
+                ->stop();
+        }
```

Comment:
I've been thinking about some of the earlier discussion about this `config->reporting()` check vs. the other places where the ptr is just checked for null. I think it can work and be clear with some simple changes. Here's a commit: https://github.com/ximinez/rippled/commit/8d963bad88f2af783a82f75aa36d428ea407f828